unsloth/kernels/utils.py

```diff
@@ -160,7 +200,7 @@ def get_lora_parameters_bias(proj):
     )
 pass
 
-if HAS_CUDA_STREAM:
+if DEVICE_TYPE == "cuda" and HAS_CUDA_STREAM:
```

Comment:
`if DEVICE_TYPE == "cuda" and HAS_CUDA_STREAM`

I don't think this is quite right. the xpu device type would fallback to the else below and would not work.

Reply:
The PR description indicates that quantization with bitsandbytes would actually be coming later.

> For the first step we are aiming to support several models with LoRA, and increase our feature in the future (including BNB, FlashAttention, xformers).